pyiron_base/job/util.py

```diff
@@ -113,12 +113,36 @@ def _get_project_for_copy(job, project, new_job_name):
     return file_project, hdf5_project
 
 
+_special_symbol_replacements = {
+    '.': 'd',
```

Comment:
I suggest that instead of having single characters replacing minus or plus sign, we use `_plus_` and `_minus_`, since `p` and `c` can be confusing with pressure and concentration. `.` can also be replaced by `_point_`. This is just a suggestion from my side. Other than that it looks good to me.

Reply:
In principle I would agree, however, we have a 50 character limit 😅

Reply:
I have exactly the same concern as @niklassiemer (and indeed in principle I agree that the replacement should be more explicit). But maybe rather than focusing on making the names more explicit, we can make both this and the inverse conversion available to the user, i.e. also a function that turns `'job_300d0_K_0_GPa_3d52_3d52_3d52'` into `['job', 300.0, 'K', 0, 'GPa', 3.52, 3.52, 3.52]`, which I think is feasible since `d`, `m` etc. appear only next to a number of an underscore.

Reply:
There, I would rather bring up the idea of metadata for all kinds of pyiron objects. This way we just store the initial list instead of trying to back-convert.

Reply:
You are right :+1:

Reply:
> 

yeah in principle I'd agree, but then often there's the reality that comes in between that it's much faster to screen the job names than loading the meta data, or sometimes it's a value that the user defines (like strain, which wouldn't be stored in pyiron), in which case there's the double work of defining the name and storing the data in the metadata for the user.

Reply:
I meant to automatically store the provided data for the name generation and return this on request. Otherwise, I fear we would have a hard time to convert back properly, e.g. I would expect some 'failures' like here:
```python
_get_safe_job_name(['some very', 300.5, 'T', 'bad.example'])
>>> 'some_very_300d5_T_baddexample' 
_revert_get_safe_job_name('some_very_500d5_T_baddexample')
>>> ['some', 'very', 300.5, 'T', 'baddexample']
```

Reply:
ah I see. That's a good idea :)

Reply:
On the other hand, if you're back converting data like this, it means we're not storing enough data in the proper places.  E.g. temperatures, k-points, number of iterations, etc. should all be present in `job['input']` and therefore accessible e.g. via pyiron tables.

Reply:
The problem is that there are some time human concepts in the input, which don't exist in atomistic simulations as such, e.g. grain boundaries, dislocations, vacancies etc. So when they are input values, like the sigma value of a grain boundary or the dislocation structure, we have to rely on the input that we provide ourselves.